whois-api/src/main/java/net/ripe/db/whois/api/rest/HttpsAuthHeaderFiler.java

```diff
@@ -0,0 +1,47 @@
+package net.ripe.db.whois.api.rest;
+
+import com.google.common.net.HttpHeaders;
+import jakarta.servlet.Filter;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.ServletRequest;
+import jakarta.servlet.ServletResponse;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletRequestWrapper;
+import jakarta.servlet.http.HttpServletResponse;
+import net.ripe.db.whois.common.oauth.OAuthUtils;
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.lang3.StringUtils;
+import org.eclipse.jetty.http.HttpStatus;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+
+import static jakarta.servlet.http.HttpServletRequest.BASIC_AUTH;
+
+@Component
+public class HttpsAuthHeaderFiler implements Filter {
+
+    @Override
+    public void doFilter(final ServletRequest request, final ServletResponse response, final FilterChain chain) throws IOException, ServletException {
+
+        if(!(request instanceof HttpServletRequest httpRequest)) {
+            chain.doFilter(request, response);
+            return;
+        }
+
+        if (StringUtils.isNotEmpty(httpRequest.getHeader(HttpHeaders.AUTHORIZATION)) && RestServiceHelper.isHttpProtocol(httpRequest)){
```

Comment:
Question:
We will apply this filter always when hitting `/whois/` 
There are multiple authenticate method that doesn't require the usage of `Authorization` header like PGP key
https://docs.db.ripe.net/Authorisation/Using-the-Authorisation-Methods/

Reply:
Yes it will always hit 

Document says :  

PGP : When using the HTTP POST and GET requests for Syncupdates, the signed update must be supplied in the message body and PGP-signed updates cannot be submitted using the RESTful API.

MD5 Password: With the RESTful API, any required passwords must be included in the URI query parameters. 

X.509 signed updates:  When using the HTTP POST and GET requests for Syncupdates, the signed update must be supplied in the message body.

X.509 signed updates cannot be submitted via the RESTful API.